app/src/main-process/main.ts

```diff
@@ -272,19 +273,45 @@ app.on('ready', () => {
 
   ipcMain.on(
     'show-contextual-menu',
-    (event: Electron.IpcMessageEvent, items: ReadonlyArray<any>) => {
+    (event: Electron.IpcMessageEvent, items: ReadonlyArray<IMenuItem>) => {
       const menu = new Menu()
-      const menuItems = items.map((item, i) => {
-        return new MenuItem({
-          label: item.label,
-          click: () => event.sender.send('contextual-menu-action', i),
-          type: item.type,
-          enabled: item.enabled,
-        })
-      })
 
-      for (const item of menuItems) {
-        menu.append(item)
+      for (const [ix, item] of items.entries()) {
+        // Special case editMenu in context menus. What we
+        // mean by this is that we want to insert all edit
+        // related menu items into the menu at this spot, we
+        // don't want a sub menu
+        if (item.role === 'editMenu') {
+          const editMenu = Menu.buildFromTemplate([item]).items[0]
+
+          if (editMenu.submenu) {
+            for (const editItem of editMenu.submenu.items) {
```

Comment:
Let's use `filter` here since that seems to be the intent.

```ts
// We don't use styled inputs anywhere at the moment
// so let's filter this for now and when/if we do we
// can make it configurable from the callee
editMenu.submenu.items
  .filter(editItem =>
    editItem.role &&
    editItem.role.toLocaleLowerCase() !== 'pasteandmatchstyle'
  )
  .forEach(editItem => menu.append(editItem))
```